src/mame/tvgames/xavix.h

```diff
@@ -854,25 +921,51 @@ class xavix_cart_state : public xavix_state
 
 protected:
 
+	virtual void xavix_extbus_map(address_map &map) override ATTR_COLD;
+
+	u8 cart_r(offs_t offset)
+	{
+		if (m_cartslot->has_cart())
+		{
+			return m_cartslot->read_cart(offset);
+		}
+		else
+		{
+			return m_rgn[(offset) & (m_rgnlen - 1)];
+		}
+	}
+
+	void cart_w(offs_t offset, uint8_t data)
+	{
+		if (m_cartslot->has_cart())
+		{
+			m_cartslot->write_cart(offset, data);
+		}
+		else
+		{
+			logerror("%s: unhandled write access to cart area with no cart installed %08x %02x\n", machine().describe_context(), offset, data);
+		}
+	}
```

Comment:
To avoid taking the performance hit on every memory access, can you install the cartridge ROM on start if a cartridge is loaded?

Reply:
I'm aware of that, this is why I've started refactoring it, again that's for a future PR, because I don't want this one to be too disruptive.